File: rust/testcode/xss_013.rs
//! CWE-79: User input placed in onclick event handler attribute.

// vuln-code-snippet start testcodeXss013
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let action = req.param("action");

    let html = format!(
        "<html><body><button onclick='doAction(\"{}\")'>Go</button></body></html>",
        action
    ); // vuln-code-snippet target-line testcodeXss013

    super::shared::BenchmarkResponse::ok(&html)
}
// vuln-code-snippet end testcodeXss013

File: php/testcode/xss_013.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_xss_href_javascript
function xss_href_javascript(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('url');

    $html = '<a href="' . $input . '">Click here</a>'; // vuln-code-snippet vuln-line php_xss_href_javascript

    return BenchmarkResponse::html($html);
}
// vuln-code-snippet end php_xss_href_javascript
